src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -1115,18 +1147,24 @@ private static unsafe int WriteFileNative(IntPtr hFile, byte[] bytes, int offset
                 bool writeSuccess;
                 fixed (byte* p = bytes)
                 {
-                    int numBytesWritten;
-                    writeSuccess = (0 != Interop.mincore.WriteFile(hFile, p + offset, count, out numBytesWritten, IntPtr.Zero));
-                    Debug.Assert(!writeSuccess || count == numBytesWritten);
-
-                    // If the code page could be Unicode, we should use ReadConsole instead, e.g.
-                    // // Note that WriteConsoleW has a max limit on num of chars to write (64K)
-                    // // [http://msdn.microsoft.com/en-us/library/ms687401.aspx]
-                    // // However, we do not need to worry about that because the StreamWriter in Console has
-                    // // a much shorter buffer size anyway.
-                    // Int32 charsWritten;
-                    // writeSuccess = Interop.mincore.WriteConsole(hFile, p + offset, count / BytesPerWChar, out charsWritten, IntPtr.Zero);
-                    // Debug.Assert(!writeSuccess || count / BytesPerWChar == charsWritten);
+                    if (useFileAPIs)
+                    {
+                        int numBytesWritten;
+                        writeSuccess = (0 != Interop.mincore.WriteFile(hFile, p + offset, count, out numBytesWritten, IntPtr.Zero));
+                        Debug.Assert(!writeSuccess || count == numBytesWritten);
```

Comment:
We throw an exception in write as well in case the error is ERROR_NO_DATA or ERROR_BROKEN_PIPE same as read.